Tests/UnitTests/Core/TrackFinding/CombinatorialKalmanFilterTests.cpp

```diff
@@ -55,8 +55,8 @@ struct ExtendedMinimalSourceLink {
       size_t sid, const FittableMeasurement<ExtendedMinimalSourceLink>* m)
       : sourceID(sid), meas(m) {}
 
-  const Surface& referenceSurface() const {
-    return *MeasurementHelpers::getSurface(*meas);
+  GeometryIdentifier geometryId() const {
+    return MeasurementHelpers::getSurface(*meas)->geometryId();
```

Comment:
Do we need to explicitly define the equality operators for the `ExtendedMinimalSourceLink` as well?

Reply:
Yes, we do. The `ExtendedMinimalSourceLink` is a separate type from the `MinimalSourceLink` in the core. The will be unified into a single `TestSourceLink` in #533.